fabtests/prov/efa/src/efa_rnr_shared.c

```diff
@@ -77,9 +77,13 @@ int ft_efa_rnr_init_fabric()
 	}
 
 	fprintf(stdout, "Setting RNR retry count to %zu ...\n", rnr_retry);
-	ret = fi_setopt(&ep->fid, FI_OPT_ENDPOINT, FI_OPT_EFA_RNR_RETRY, &rnr_retry, sizeof(rnr_retry));
+	/* 
+	 * Cannot use setopt FI_OPT_EFA_RNR_RETRY here because it sets FI_RM_DISABLED, 
+	 * which conflicts with the queue/resend test that requires FI_RM_ENABLED.
+	 */
+	ret = setenv("FI_EFA_RNR_RETRY", "0", 1);
 	if (ret) {
-		FT_PRINTERR("fi_setopt", -ret);
+		FT_PRINTERR("setenv", -ret);
```

Comment:
I don't think setenv works here because `fi_param_get` happens at fi_getinfo where we already have a efa_env.rnr_retry set as 3.

Also, this is shared among the rnr_cq_read_error and queue_resend test. The former intends to turn off RM, the latter one intends to turn on RM. It's totally fine for rnr_cq_read_error to call fi_setopt, but not queue_resend.

So we should have a bool somewhere to call fi_setopt only for cq_read_error test. And make queue_resend_test setenv before initing any resources

Reply:
Looking at the code again, Nah. rnr_retry i not really environment variable even though it is in `efa_env` so setenv won't work at all.